components/TextInput.js

```diff
@@ -1,28 +1,69 @@
-import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
+import autoBindReact from 'auto-bind/react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
 import { TextInput as RNTextInput } from 'react-native';
 
-import { connectStyle } from '@shoutem/theme';
 import { connectAnimation } from '@shoutem/animation';
+import { connectStyle } from '@shoutem/theme';
+
+import { Caption } from './Text';
+import { View } from './View';
+
+// Style properties defined in theme which would case a warning if passed to a components style prop
+const omittedStyleProperties = [
+  'errorBorder',
+  'placeholderTextColor',
+  'selectionColor',
+  'underlineColorAndroid',
+  'withBorder',
+  'withoutBorder',
+];
 
 class TextInput extends PureComponent {
-  render() {
-    const { props } = this;
-    const style = {
-      ...props.style,
+  constructor(props) {
+    super(props);
+
+    autoBindReact(this);
+
+    this.state = {
+      isFocused: props.highlightOnFocus && props.autoFocus,
     };
-    delete style.placeholderTextColor;
-    delete style.selectionColor;
-    delete style.underlineColorAndroid;
+  }
+
+  handleBlur() {
+    this.setState({ isFocused: false });
+  }
+
+  handleFocus() {
+    this.setState({ isFocused: true });
+  }
+
+  render() {
+    const { errorMessage, highlightOnFocus, style } = this.props;
+    const { isFocused } = this.state;
 
     return (
-      <RNTextInput
-        {...props}
-        style={style}
-        placeholderTextColor={props.style.placeholderTextColor}
-        selectionColor={props.style.selectionColor}
-        underlineColorAndroid={props.style.underlineColorAndroid}
-      />
+      <View>
+        <RNTextInput
+          {..._.omit(this.props, 'style')}
+          onBlur={this.handleBlur}
+          onFocus={this.handleFocus}
+          placeholderTextColor={style.placeholderTextColor}
+          selectionColor={style.selectionColor}
+          underlineColorAndroid={style.underlineColorAndroid}
+          style={[
+            _.omit(style, omittedStyleProperties),
```

Comment:
Again, I'd opt here for the approach: `const { placeholderTextColor, selectionColor, ..., ...stylesheetStyle } = style;` 
And then you'd say `placholderTextColor={placeholderTextColor}` (and so on for the rest of them) and `style={...stylesheetStyle, ...}`

I'm not a fan of this `omittedStyleProperties` array because we're gonna have to maintain it.